node/core/pvf-checker/src/metrics.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Metrics definitions for the PVF pre-checking subsystem.
+
+use polkadot_node_subsystem_util::metrics::{self, prometheus};
+
+#[derive(Clone)]
+struct MetricsInner {
+	pre_check_judgement: prometheus::Histogram,
+	votes_total: prometheus::Counter<prometheus::U64>,
+	votes_started: prometheus::Counter<prometheus::U64>,
+	votes_duplicate: prometheus::Counter<prometheus::U64>,
```

Comment:
Generally curious, can you elaborate why `CounterVec` is more appropriate here than a `Counter`?

Reply:
That's my personal choice, it's fine either way. I like it more because reduces the number of metric definitions, but you still have 3 metrics tracked . Then you can write a shorter grafana metrics query when grouping by labels.

Reply:
Ok, I see what you mean now. I wasn't sure what was `CounterVec` but that's just a counter with different labels. I agree that it would be a better approach, but only marginally better.

However, I am trying to make it to the next release if possible. So maybe I can take care of that as a follow-up